python/CMakeLists.txt

```diff
@@ -1,6 +1,6 @@
-cmake_minimum_required(VERSION 3.16)
+cmake_minimum_required(VERSION 3.18)
```

Comment:
Is this version high enough for `scikit-build-core`?

Reply:
Yes, scikit-build-core requires cmake 3.15 or later. I copied 3.18 from the C++ layer.